---
title: "Coverage plots"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

The coverage plots produced below correspond to the part of the genome with coverage (excluding 0-coverage regions that for most account for N-regions. **0-coverage** regions are measured for each chromosome and returned in a table with the relative fraction of the chromosome in the last column.

Some chromosome models represent alt-alleles (alt-) or hetetochromatine (het-) and are not necessarily expected to attract reads, especially when a repeat-mask was applied to the genome prior to building the mapper index.

```{r}
library("knitr")
library("plotrix")

path <- "/data1/SRR833244"
#setwd(path)
opts_knit$set(root.dir = path)

folder <- "BDGP5.78_bedtools_genomeCvg"
file <- paste(folder, "BDGP5.78_bwa-mapping-histo.txt", sep="/")
name <- "BDGP5.78_bwa"

outfolder <- paste(folder, "/", name, "-coverage_plots", sep="")
dir.create(outfolder)
```

```{r, as.is=TRUE}
data <- read.delim(file, header=F)
colnames(data) <- c("chr", "depth", "count", "chrlen", "cov%")

chr.list <- as.vector(unique(data$chr))

for(chr in chr.list){
  subset <- data[data$chr==chr,]

  # find X-coordinates with 0-coverage
  zero <- subset[subset$depth==0,]
  cat(paste("# fraction of the chromosome:", chr, "with zero-coverage\n", sep=" "))
  print(zero)
  cat("\n\n")

  # find X-coordinates for 5%, Q1, Q3, 95%
  sub <- subset[subset$depth>0,]
  
  # print summary
  cat(paste("# coverage summary for chromosome:", chr, "\n", sep=" "))
  print(summary(sub))
  cat("\n\n")
  
  vec <- rep(sub$depth, sub$count)
  Q001 <- quantile(vec, 0.01)
  Q005 <- quantile(vec, 0.05)
  Q1 <- summary(vec)[2]
  Q2 <- summary(vec)[3]
  Q3 <- summary(vec)[5]
  Q95 <- quantile(vec, 0.95)

  # create table for plot
  leg.table <- data.frame(depth=c(Q001, Q005, Q1, Q2, Q3, Q95))

  # create pictures with coverage distribution
  title <- paste("coverage depth (", name, "-", chr, ")", sep="")

  # limits for all and sample datasets
  xmax=2*Q2
  max.count <- max(sub[sub$depth<xmax,]$count)
  ymax <- max.count
  plot(sub$count ~ sub$depth, 
  xlim=c(0, xmax),
  ylim=c(0, ymax),
  xlab=title,
  ylab="bp-count", 
  type="l",
  col="blue",
  lwd=3,
  cex=0.5)

  # add vertical line at pic
  abline(v=Q001, col="grey", lwd=1)
  abline(v=Q005, col="grey", lwd=2)
  abline(v=Q1, col="grey", lwd=2)
  abline(v=Q2, col="red", lwd=2)
  abline(v=Q3, col="grey", lwd=2)
  abline(v=Q95, col="grey", lwd=2)

  # show most of the options
  addtable2plot((70/100*xmax), 
    (60/100*ymax), 
    leg.table,
    bg="white",
    bty="o", 
    display.rownames=TRUE,
    hlines=TRUE, 
    vlines=TRUE,
    cex=1)
}
```
